src/controllers/document.controller.ts

```diff
@@ -38,7 +44,8 @@ export default async function (fastify: FastifyInstance) {
       return reply.fail({ role: 'missing permission' }, 403)
 
     try {
-      const repo = new Document(request.log)
+      const pool = await fastify.getSqlPool(env['DB_NAME'] ?? 'PCM')
```

Comment:
Maybe default to DB_NAME in getSqlPool, and allow you to configure a default DB in the Fastify wrapper config.

This should reduce duplicate code like this